🐛 Sets airbyte-cdk on source-netsuite to use version 0.90.0

tcboles opened the pull request:
## What
 source-netsuite depends on code from airbyte-cdk pre v1.0.0. 

## How
By setting the version of airbyte-cdk to 0.90.0, the connector works

## Review guide
Try to source-netsuite with pyairbyte to confirm a failed connector
Set pip_url on the get_source function to the local source-netsuite with pinned version and try the connector again 

## User Impact
There should be no user impact besides a working connector

## Can this PR be safely reverted and rolled back?
YES 💚

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=44189) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=44189) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=44189) it.</sub>

natikgadzhi reviewed (commented):
Thank you for putting this together! To merge this, we'd need to migrate this to Poetry instead of replacing in `setup.py`.

To do this: 
- Delete `setup.py`
- Make a `pyproject.toml` (see examples in other connectors)
- `poetry install` so it makes the lockfile 
- Update the readme
- Bump versions in `pyproject.toml` and `metadata.yaml`.

tcboles commented:
> Thank you for putting this together! To merge this, we'd need to migrate this to Poetry instead of replacing in `setup.py`.
> 
> To do this:
> 
> * Delete `setup.py`
> * Make a `pyproject.toml` (see examples in other connectors)
> * `poetry install` so it makes the lockfile
> * Update the readme
> * Bump versions in `pyproject.toml` and `metadata.yaml` and the docs page.
> 
> Would you be interested in fixing this up?

Absolutely. I will get an update over in a few minutes.

tcboles commented:
@natikgadzhi I have added poetry and updated the readme. I am confused why `Connectors early CI checks` is failing. When I click on details, it says that the 3 items that ran passed and 2 items are still waiting approval.

natikgadzhi commented:
Thank you for the cleanup! Let me kick off the CI and see what's up. 

For my understanding, can you build this with all the cleanup into a custom connector and confirm it works and doesn't loose data for your use case?

natikgadzhi commented:
/format-fix

natikgadzhi commented:
The changes themselves look fine, but the build system is not happy.

natikgadzhi commented on airbyte-integrations/connectors/source-netsuite/.tool-versions:
```diff
@@ -0,0 +1 @@
+python 3.11.9
```
Remove this, please. Hypothetically no harm, but mostly we target Python 3.10+, and we don't want anything that won't run on 3.10 to make it into any given connector just yet.

tcboles replied:
Thanks for catching this! I was trying to keep it out of my commits but must have done an add all.

natikgadzhi commented on airbyte-integrations/connectors/source-netsuite/metadata.yaml:
```diff
@@ -35,4 +35,6 @@ data:
           secretStore:
             type: GSM
             alias: airbyte-connector-testing-secret-store
+  connectorBuildOptions:
```
That's cool, thank you for switching!

natikgadzhi commented:
Re-started CI. Aside from python version set to 3.11 in `tools-versions` this is good.

natikgadzhi commented on airbyte-integrations/connectors/source-netsuite/CHANGELOG.md:
```diff
@@ -1,9 +0,0 @@
-# Changelog
-
```
It's unlikely that we would actually maintain a changelog, so that thing is probably best removed. Thank you for the cleanup.

natikgadzhi commented:
Alright, so, I think we're hitting the reason why we haven't updated it in the first place — CATs (tests) are not behaving well, here's what we see (we'll make the reports public soon).

```
1 new failing acceptance tests detected:
-test_core.py::TestDiscovery::test_discover[inputs0]
-test_incremental.py::TestIncremental::test_two_sequential_reads[inputs0]
-test_core.py::TestDiscovery::test_defined_cursors_exist_in_schema[inputs0]
-test_core.py::TestDiscovery::test_primary_keys_exist_in_schema[inputs0]
-test_core.py::TestDiscovery::test_defined_refs_exist_in_schema[inputs0]
-test_core.py::TestDiscovery::test_defined_keyword_exist_in_schema[inputs0-not]
-test_core.py::TestDiscovery::test_primary_keys_data_type[inputs0]
-test_core.py::TestDiscovery::test_streams_has_sync_modes[inputs0]
-test_core.py::TestDiscovery::test_additional_properties_is_true[inputs0]
-test_core.py::TestDiscovery::test_defined_keyword_exist_in_schema[inputs0-allOf]
-test_core.py::TestDiscovery::test_backward_compatibility[inputs0]
-test_core.py::TestBasicRead::test_read[inputs0]
-test_core.py::TestConnection::test_check[inputs0]
-test_full_refresh.py::TestFullRefresh::test_sequential_reads[inputs0]
-test_core.py::TestConnection::test_check[inputs1]
-test_incremental.py::TestIncremental::test_state_with_abnormally_large_values[inputs0]
-test_incremental.py::TestIncremental::test_read_sequential_slices[inputs0]
-test_core.py::TestDiscovery::test_streams_have_valid_json_schemas[inputs0]
Please fix the new failing tests before merging this PR.
Please also check the original 18 acceptance tests failures and assess how critical they are.
```

It could be that our credentials just expired. If you're open to shareing your netsuite config with me in a secure way (ping me on community Slack, we'd figure out how to transfer) or can provide a PDF of your test report from `airbyte-ci connectors --name source-netsuite test` with your creds performed locally, that would bump my confidence level.

tcboles commented:
> Alright, so, I think we're hitting the reason why we haven't updated it in the first place — CATs (tests) are not behaving well, here's what we see (we'll make the reports public soon).
> 
> ```
> 1 new failing acceptance tests detected:
> -test_core.py::TestDiscovery::test_discover[inputs0]
> -test_incremental.py::TestIncremental::test_two_sequential_reads[inputs0]
> -test_core.py::TestDiscovery::test_defined_cursors_exist_in_schema[inputs0]
> -test_core.py::TestDiscovery::test_primary_keys_exist_in_schema[inputs0]
> -test_core.py::TestDiscovery::test_defined_refs_exist_in_schema[inputs0]
> -test_core.py::TestDiscovery::test_defined_keyword_exist_in_schema[inputs0-not]
> -test_core.py::TestDiscovery::test_primary_keys_data_type[inputs0]
> -test_core.py::TestDiscovery::test_streams_has_sync_modes[inputs0]
> -test_core.py::TestDiscovery::test_additional_properties_is_true[inputs0]
> -test_core.py::TestDiscovery::test_defined_keyword_exist_in_schema[inputs0-allOf]
> -test_core.py::TestDiscovery::test_backward_compatibility[inputs0]
> -test_core.py::TestBasicRead::test_read[inputs0]
> -test_core.py::TestConnection::test_check[inputs0]
> -test_full_refresh.py::TestFullRefresh::test_sequential_reads[inputs0]
> -test_core.py::TestConnection::test_check[inputs1]
> -test_incremental.py::TestIncremental::test_state_with_abnormally_large_values[inputs0]
> -test_incremental.py::TestIncremental::test_read_sequential_slices[inputs0]
> -test_core.py::TestDiscovery::test_streams_have_valid_json_schemas[inputs0]
> Please fix the new failing tests before merging this PR.
> Please also check the original 18 acceptance tests failures and assess how critical they are.
> ```
> 
> It could be that our credentials just expired. If you're open to shareing your netsuite config with me in a secure way (ping me on community Slack, we'd figure out how to transfer) or can provide a PDF of your test report from `airbyte-ci connectors --name source-netsuite test` with your creds performed locally, that would bump my confidence level.

Thanks @natikgadzhi. Unfortunately the creds we are using are using are scoped down and are from a client, so they cannot be shared.I rant the tests locally, but it fails since we do not have access to all of the objects needed for the test.

natikgadzhi commented:
Understandable. 

I can build a pre-release version based on this to verify that works for your case at the very least, then see how to best test it. 

@girarda we care if this breaks CATs or nah?

girarda commented:
@natikgadzhi if this is a credentials problem, I'd expect CATs to be failing on master. If that's the case ✅

tcboles commented:
Hi, @natikgadzhi and @girarda. Running the connector locally I ended up running into a few issues with how the native rate limit handler worked with netsuite since its using Oauth1. It would try to resend the request, but the request non longer had a valid signature causing netsuite to then throw a 401. I added backoff logic to the source itself and it now handles that issue. 

Do you still see test failures on your end now that this change is in place?

natikgadzhi commented:
I've merged master into this, will rerun tests. As long as basic unit tests are good, I'll kick-off a progressive rollout.
